### PR TITLE
To add download csv

### PR DESCRIPTION
**Pull request by arosgood**
Just adding something to give people the option of downloading their data to CSV. Let me know what you guys think!

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -286,11 +287,15 @@ class App extends React.Component {
         isLoading={this.getIsLoadingDataset('dwells')}
         data={this.state.dwells}
         xField={'arr_dt'}
-        xFieldLabel={'Time of day'}
+        xFieldLabel={'Time of day'}
```
Comment (nathan-weinberg) [inline src/App.js:290]: nit: can we remove the whitespace here?

**Review by nathan-weinberg — APPROVED** [review]
Code LGTM, can we get a visual of what it looks like @arosgood?

File: src/App.js (hunk 1 above)
Comment (arosgood) [reply src/App.js]: Yes haha my bad, removed it.

**Comment by arosgood** [comment]
> Code LGTM, can we get a visual of what it looks like @arosgood?

Here's the visual:
Screen Shot 2021-02-22 at 10 51 23 PM

**Comment by mathcolo** [comment]
Hey @arosgood in letting this percolate more, I am wondering if folks might be confused that all three metrics are in the same CSV. Not sure though?

**Comment by friendchris** [comment]
[dashboard_data.xlsx](https://github.com/transitmatters/t-performance-dash/files/6046196/dashboard_data.xlsx)

I think it would be good to have each chart download separately, if possible. I've attached an Excel with how I think each one should look. Let me know if you all think this makes sense.

**Comment by nathan-weinberg** [comment]
I like the look of the link but I'm wondering if we can make it part of the footer?

**Comment by arosgood** [comment]
Can a csv have multiple sheets like that? I could also make it a .xlsx instead but maybe that's less accessible for people? I was also thinking about adding it to the footer, but Andrew recommended having it above the footer. Maybe it's better to have each metric separate?

**Comment by arosgood** [comment]
New visual:
image

File: src/line.js [hunk 2]
```diff
@@ -38,13 +42,16 @@ const point_colors = (data, metric_field, benchmark_field) => {
...
     else if (ratio <= 1.5) {
-      return '#e5a70b'; //yellow
+      return '#f5ed00'; //yellow
+    }
+    else if (ratio <= 2.0) {
+      return '#c33149'; //red
     }
-    else if (ratio > 1.5) {
-      return '#e53a0b'; //red
+    else if (ratio > 2.0) {
```
Comment (mathcolo) [inline src/line.js:53]: Is this branch based on an old revision possibly? My guess is that you aren't intending to change these colors?

File: src/line.js [hunk 3]
```diff
@@ -3,6 +3,8 @@ import classNames from 'classnames';
 import { Line, Chart } from 'react-chartjs-2';
 import Legend from './Legend';
 import drawTitle from './Title';
+import { CSVLink } from "react-csv";
```
Comment (mathcolo) [inline src/line.js:6]: Should be fine to import these libraries, but we probably need to add attribution to the open source page at https://github.com/transitmatters/t-performance-dash/blob/main/src/OpenSource.js !

File: src/line.js [hunk 4]
```diff
@@ -83,12 +90,21 @@ class LineClass extends React.Component {
     //   }
     // });
     const { isLoading } = this.props;
+
+    const graphData=   
+    <span className="graphData">
```
Comment (mathcolo) [inline src/line.js:95]: Might be cleaner if indented in?

**Review by mathcolo — COMMENTED** [review]
I think this is close! added a few more comments

File: src/line.js (hunk 2 above)
Comment (arosgood) [reply src/line.js:53]: ah yea mb

File: src/line.js (hunk 3 above)
Comment (arosgood) [reply src/line.js:6]: ok will do ty

File: src/line.js (hunk 4 above)
Comment (arosgood) [reply src/line.js:95]: yes, will change

**Comment by nathan-weinberg** [comment]
@mathcolo @arosgood Looks like this one fell through the cracks a bit, we still interested in getting it in?

**Comment by ianhi** [comment]
Hi,

What more needs to happen to get this in? This would be great.

**Comment by nathan-weinberg** [comment]
Hi @ianhi,

So we discussed this internally and due to the age of this PR we're going to need to close it and refactor the code - if you're interested in helping out please shoot us an email at labs@transitmatters.org and we would be happy to get you in the loop!

\- Nathan

**Pull request closed without merging**